app/controllers/api/host_initiators_controller.rb

```diff
@@ -0,0 +1,41 @@
+module Api
+  class HostInitiatorsController < BaseController
+    def refresh_resource(type, id, _data = nil)
+      raise BadRequestError, "Must specify an id for refreshing a #{type} resource" if id.blank?
+
+      ensure_resource_exists(type, id) if single_resource?
+
+      api_action(type, id) do |klass|
+        host_initiator = resource_search(id, type, klass)
+        api_log_info("Refreshing #{host_initiator_ident(host_initiator)}")
+        refresh_host_initiator(host_initiator)
+      end
+    end
+
+    def create_resource(_type, _id = nil, data = {})
+      ext_management_system = ExtManagementSystem.find(data['ems_id'])
```

Comment:
Since we're using ems_id, probably check for its existence before using it and throw a BadRequestError accordingly.   Maybe add a test for that case.

Reply:
@abellotti I added the check for ems_id existence and test for it.